### ISSUE TITLE
All repositories reported as missing

### ISSUE THREAD

**Issue by jairoalves** [desktop/desktop#11620]
### Describe the bug

Git Desktop can't find any of my existing repositories. I cannot commit to any one of them, app reports it can't find them, but they are all present and git folders exist too.

### Version & OS

Git Desktop 2.6.3 on Windows 10 (1909)

### Steps to reproduce the behavior

1. Go to Current Repository selector
2. Click on any one of the repositories on the list.
3. See the error

### Expected behavior

Expected the repository to be selected and its changes to be displayed for commit.

### Actual behavior

Repository won't load, says it `Can't find "<repo_name>"`. Also, it shows a `It was last seen at "<repo_path>" ` message.

### Screenshots

![image](https://user-images.githubusercontent.com/29930394/108356001-4cc14880-71ca-11eb-9834-079c403d4267.png)

### Logs

Did not locate related errors in the logs, attached below.

[2021-02-18.desktop.production.log](https://github.com/desktop/desktop/files/6002785/2021-02-18.desktop.production.log)

### Additional context

Tried running `git status` on the command line, to no success.
```C:\Users\myUserFolder\OneDrive - My company\Documents\GitHub>C:\Users\myUserFolder\AppData\Local\GitHubDesktop\app-2.6.3\resources\app\git\cmd\git.exe status
fatal: not a git repository (or any of the parent directories): .git
```

It says `not a git repository` although the git folder exists, and also does not seem to be corrupted.

Tried deleting an unchanged local repo e re-cloning it, it works (for a while).

```Cloning into 'VMS.APACHE'...
remote: Counting objects: 70, done.
remote: Compressing objects: 100% (67/67), done.
remote: Total 70 (delta 33), reused 0 (delta 0)
Unpacking objects: 100% (70/70), 391.15 KiB | 73.00 KiB/s, done.
Updating files: 100% (57/57), done.
```

But after some seconds, it will fall into the same `can't find "repo_name"` error.

Also tried reinstalling the Git Desktop App, it did not seem to help.

**Comment by sergiou87** [comment]
Hi @jairoalves! 👋

I'm sorry you're having this issue 😞  If I'm reading your logs correctly, seems like you have your repos (or at least a few of them) inside of a OneDrive folder? If so, that is known to cause some issues.

Can you try resetting GitHub Desktop by following these steps and see if it helps?

1. Close GitHub Desktop 
2. Open File Explorer and remove all of the files in `%APPDATA%\GitHub Desktop\`
3. Restart GitHub Desktop (you will need to log in and add your repositories again)

**Comment by jairoalves** [comment]
Dear, Sergio, first of all, thanks for helping me out! 

I deleted all files and folders below the GitHub Desktop folder as indicated (see image below).
![image](https://user-images.githubusercontent.com/29930394/108392775-c91b5200-71f1-11eb-8392-b92135273ce7.png)

Then, after the restart of GitHub Desktop, I clicked on the "Add local repository" function.
As I enter / select the local folders where my repos ares, GitHub Desktop seems to not recognize the local repositories as such, as seen on the image below.
![image](https://user-images.githubusercontent.com/29930394/108393728-b35a5c80-71f2-11eb-9735-360306c7c20b.png)

For testing purposes, I've downloaded the GitKraken client. I've noticed that it can recognize the local repos. I tested stashing and pushing to a couple of repos, that worked as well.

.

Then, also for testing, I tried cloning a repo from `origin`. Renamed my local folder and cloned the master.
It seemed to work ok, but a little later I did my first commit, the same error was generated once again, as in the image below.

![image](https://user-images.githubusercontent.com/29930394/108397784-177f1f80-71f7-11eb-88fc-539060d12927.png)

**Comment by steveward** [comment]
@jairoalves you can drag-and-drop multiple repositories into GitHub Desktop to add them. That may be quicker if you need to add a number of repositories.

 We have seen cases like this in the past where repositories have been lost after updating (see https://github.com/desktop/desktop/issues/10548). Did this happen after the application updated?

**Comment by jairoalves** [comment]
yes, I noticed the issue roughly after the last auto-update. For the record, a few days before the update, I executed a script that dealed with some 4500+ files on the repo folder, maybe that could have triggered some gitignore or status-size limits, idk. That being said, if I recall correctly, it was working fine after the 4500 files thing but not anymore after the upgrade.

**Comment by jairoalves** [comment]
So, an important update. I've just removed every repo from any Onedrive folder. Tried adding again as a local existing repo and this time it is working.

**Comment by jairoalves** [comment]
After reseting the configuration and removing the local repos from Onedrive, it is working normal as far as I can tell. Should I close the issue?

**Comment by steveward** [comment]
@jairoalves I'm glad to hear that things are working again. We can close this issue out, and I'll reference this issue in the issue I linked above so that we can track these cases.

**Comment by JimHokanson** [comment]
I had to do some command line work so I recently added a SSH key and installed Git for Windows and did some global config edits (Using windows git client, what I thought was the same as my GitHub account info). Suddenly many of my private repos were not found (repos I had not touched with the git command line) and when I tried to manually add them GitHub Desktop said they weren't git repos. Fortunately, I updated GitHub desktop and then it prompted with with warnings for these repos - something to the effect of the repo might be unsafe and that it is from another user and wanted to know if I should add the repo to a trusted list. Fortunately that seems to have fixed the problem. Maybe the new update truly fixed the problem or maybe it reset some state. If it is the latter, I wonder the errors about a missing repo or the folder not being a repo are, in some cases, this untrusted repo case, and the error handler is too generic ... (i.e., something went wrong, repo must be missing).
